specification/sql/resource-manager/Microsoft.Sql/preview/2020-08-01-preview/DatabaseRecommendedActions.json

```diff
@@ -540,6 +535,21 @@
           "$ref": "#/definitions/RecommendedActionProperties",
           "description": "Resource properties.",
           "x-ms-client-flatten": true
+        },
+        "id": {
+          "description": "Resource ID.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "Resource name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "Resource type.",
+          "type": "string",
+          "readOnly": true
```

Comment:
Is there reason not use ref common definition any more?

Reply:
@akning-ms 

these specs files are auto-generated via internal DSMainDev repo. We found that the same  `RecommendedActionResource` in DSMainDev code base was referenced from the same entity, but in the spec file, it was generated in different ways:
https://github.com/Azure/azure-rest-api-specs/blob/3a3a9452f965a227ce43e6b545035b99dd175f23/specification/sql/resource-manager/Microsoft.Sql/preview/2020-08-01-preview/DatabaseRecommendedActions.json#L520-L546
vs
https://github.com/Azure/azure-rest-api-specs/blob/3a3a9452f965a227ce43e6b545035b99dd175f23/specification/sql/resource-manager/Microsoft.Sql/preview/2020-08-01-preview/DatabaseAdvisors.json#L581-L616 

This is a bug in DSMainDev codebase. But until now, we still are debugging to figure out the root cause. To walkaround the validation errors in this PR, we just keep all the RecommendedAction defs in the same way: https://github.com/Azure/azure-rest-api-specs/search?q=recommendedaction

Once we fix the bug in DSMainDev repo, we will push an update soon.

Reply:
use common type to replace all local defs.